src/constants.mjs

```diff
@@ -58,24 +58,31 @@ export const DOC_API_SLUGS_REPLACEMENTS = [
 // is a specific type of API Doc entry (e.g., Event, Class, Method, etc)
 // and to extract the inner content of said Heading to be used as the API doc entry name
 export const DOC_API_HEADING_TYPES = [
-  { type: 'method', regex: /^`?([A-Z]\w+(?:\.[A-Z]\w+)*\.\w+)\([^)]*\)`?$/i },
+  {
+    type: 'method',
+    regex:
+      /^`?(?:(?:(?:(?:\\?_)+|\b)\w+\b|\\?\[[\w.]+\\?\])\.?)*((?:(?:(?:\\?_)+|\b)\w+\b|\\?\[[\w.]+\\?\]))\([^)]*\)`?$/i,
```

Comment:
Also these regexes (or at least this particular one) is vulnerable to ReDos.

Reply:
I changed them to the originals one's from https://github.com/nodejs/node/blob/main/tools/doc/json.mjs because the previous one's didn't match the correct headings.

Look at <https://regex101.com/r/ssgo5D/1> vs <https://regex101.com/r/BBqLGX/1>

I'll look into resolving the ReDoS tho.

Reply:
That is interesting. Yes, let's iterate on the current ones we have to fix the gaps. Appreciate the examples!

These old (current) ones are super old and IDK who even created them.